test: topology: update run_first lists

patjed41 opened the pull request:
`run_first` lists in `suite.yaml` files provide a simple way to
shorten the tests' average running time by running the slowest
tests at first.

We update these lists, since they got outdated over time:
- `test_topology_ip` was renamed to `test_replace`
   and changed suite,
- `test_tablets` changed suite,
- new slow tests were added:
  - `test_cluster_features`,
  - `test_raft_cluster_features`,
  - `test_raft_ignore_nodes`,
  - `test_read_repair`.

kbr-scylla reviewed (commented):
This is proof that this `run_first` was not the best idea, it rots over time.

scylladb-promoter commented:
#### :red_circle: CI State: FAILURE

:white_check_mark: - Build
:x: - Sanity Tests
:white_check_mark: - Unit Tests
#### Failed Tests (1/21858):
- [test_streaming_on_repair](https://jenkins.scylladb.com//job/scylla-master/job/scylla-ci/4777/testReport/junit/compaction_additional_test/TestTimeWindowDataSegregation/Tests___Sanity_Tests___test_streaming_on_repair) [:mag:](https://github.com/scylladb/scylladb/issues?q=is:issue+is:open+test_streaming_on_repair)
#### Build Details:
- Build URL: https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/4777/
- Duration: 2 hr 25 min
- Builder: spider6.cloudius-systems.com

patjed41 commented:
> #### 🔴 CI State: FAILURE
> ✅ - Build ❌ - Sanity Tests ✅ - Unit Tests
> 
> #### Failed Tests (1/21858):
> * [test_streaming_on_repair](https://jenkins.scylladb.com//job/scylla-master/job/scylla-ci/4777/testReport/junit/compaction_additional_test/TestTimeWindowDataSegregation/Tests___Sanity_Tests___test_streaming_on_repair) [🔍](https://github.com/scylladb/scylladb/issues?q=is:issue+is:open+test_streaming_on_repair)
> 
> #### Build Details:
> * Build URL: https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/4777/
> * Duration: 2 hr 25 min
> * Builder: spider6.cloudius-systems.com

Known dtest issue

nyh commented:
Guys, @kostja I think the whole idea of a separate "suite.yaml" with configuration options was a mistake:
1. If a test is "slow" or "should be run first", this can be a tag on the test - not a "run_first" line in a separate file that becomes obsolete a minute after writing it.
2. If a test should be "skipped in release", again this should be a tag on the actual test (a variant of pytest.mark.skip), not an entry in suite.yaml.
3. If a test needs a 3-node cluster, this should be again a marker on an individual test - or a whole file - and not something in suite.yaml which forces tests that start with a 3-node cluster to live in a separate directory from tests that start with a 4-node or 0-node cluster.

kbr-scylla commented:
@nyh I mostly agree.

Although with the third point -- the reason we have a suite with pre-configured 3-node clusters, is because we wanted these clusters to get reused between tests. So there is a pool of 3-node clusters and a test takes from this pool then returns the cluster back to the pool, so it can be taken by another test.

However, this works only for a few tests, those that don't do topology operations (which mark the cluster as "dirty" -- nonreusable). It provides little value it seems. Most tests that don't do topology operations can be single node tests anyway.

kostja commented:
No point in discussing test.py design issues at random tickets. @nyh please either create a single, editable source of test.py roadmap (google doc for now only), or ... well... you can of course continue leaving comments here and htere, but that has very little chances of triggering the change.

nyh commented:
> No point in discussing test.py design issues at random tickets. @nyh please either create a single, editable source of test.py roadmap (google doc for now only), or ... well... you can of course continue leaving comments here and htere, but that has very little chances of triggering the change.

I'm not trying to "trigger change", I'm trying to change the minds of those (mostly, you and Kamil) who are actively working and writing code towards what I think is the **wrong** direction - showing them where I think they are wrong, hoping they stop pushing toward that (in my opinion) wrong direction.

nyh commented:
> Unless you mean something else by "wrong direction" in which case I don't know what you mean -- please elaborate.

(Kostja, sorry for the spam, but I'm just replying:)

I was referring to https://github.com/scylladb/scylladb/issues/16026 where I was complaining that we keep creating more and more subdirectories of test/ that differ only in different configuration in the yaml file (different number of nodes in the cluster, different parameters to start Scylla) and trying to convince the people who actively keep adding these subdirectories to stop doing this.
When I get fed up, I'll propose a patch. I'm not shy to add code in test/ - git blame tells me I wrote 53,000 lines of code there ;-)

kostja commented:
@nyh I think you overestimate my abilities. Every day I perform hundreds if not thousands of context switches. I am a rather organized person so perhaps you have an impression that I keep all of that in RAM, but no, it's swapped out pretty rapidly. You can be certain that everything we discuss at issue 16-thousand 0-something I will blank out tomorrow. I hope now I provided a clear explanation why discussions in issues is not the best idea, with my own comment showing how *far* from the original subject such a discussion can travel. 

I hope we can stop at that, and will go back to some neatly structured procedure when the feedback can be groomed later.

patjed41 commented:
v2:
- rebased,
- moved `test_replace` entry in `run_first` to `topology_custom` after this test was moved there,
- removed the paragraph about "not touching the `topology_raft_disabled` suite" from the commit message because it wasn't right -- we don't need to change `run_first` in `topology_raft_disabled` because it's up to date and not because it's unnecessary.

scylladb-promoter commented:
#### :red_circle: CI State: FAILURE

:white_check_mark: - Build
:x: - Unit Tests
:white_check_mark: - Sanity Tests
#### Failed Tests (2/21866):
- [topology_experimental_raft.test_tablets.debug.2](https://jenkins.scylladb.com//job/scylla-master/job/scylla-ci/4882/testReport/junit/%28root%29/non-boost%20tests/Tests___Unit_Tests___topology_experimental_raft_test_tablets_debug_2) [:mag:](https://github.com/scylladb/scylladb/issues?q=is:issue+is:open+topology_experimental_raft.test_tablets.debug.2)
- [test_tablet_metadata_propagates_with_schema_changes_in_snapshot_mode](https://jenkins.scylladb.com//job/scylla-master/job/scylla-ci/4882/testReport/junit/%28root%29/test_tablets/Tests___Unit_Tests___test_tablet_metadata_propagates_with_schema_changes_in_snapshot_mode) [:mag:](https://github.com/scylladb/scylladb/issues?q=is:issue+is:open+test_tablet_metadata_propagates_with_schema_changes_in_snapshot_mode)
#### Build Details:
- Build URL: https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/4882/
- Duration: 2 hr 16 min
- Builder: spider8.cloudius-systems.com

scylladb-promoter commented:
#### :green_circle: CI State: SUCCESS

:white_check_mark: - Build
:white_check_mark: - Sanity Tests
:white_check_mark: - Unit Tests
#### Build Details:
- Build URL: https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/4889/
- Duration: 2 hr 16 min
- Builder: spider7.cloudius-systems.com
